src/System.Text.JsonLab/System/Text/Json/JsonWriter.cs

```diff
@@ -413,9 +660,59 @@ private void WriteValueUtf8(ReadOnlySpan<byte> valueSpanByte, int bytesNeeded)
         /// Write a signed integer value into the current array.
         /// </summary>
         /// <param name="value">The signed integer value to be written to JSON data.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void WriteValue(long value)
         {
-            WriteValueUtf8(value, CalculateValueBytesNeeded(sizeof(byte), JsonWriterHelper.NewLineUtf8.Length));
+            int bytesNeeded = 0;
+            if (_indent < 0)
+            {
+                bytesNeeded = 1;
+            }
+
+            if (_prettyPrint)
+            {
+                bytesNeeded += JsonWriterHelper.NewLineUtf8.Length + (_indent & RemoveFlagsBitMask) * 2;
```

Comment:
nit: move this calc into `WriteValueUtf8Pretty`?

Reply:
I want to keep the bytes needed calculation in one place. This code will likely change anyway as I experiment with different approaches (for example TryWrite instead of calculating bytes needed up front).